sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs

```diff
@@ -82,5 +83,73 @@ public virtual Pageable<TableResponseProperties> GetTables(string select = null,
+            CreateTableInternalAsync(false, tableName, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Creates a table in the storage account.
+        /// </summary>
+        /// <param name="tableName">The table name to create.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns></returns>
+        [ForwardsClientCalls]
+        public virtual async Task<TableResponse> CreateTableAsync(string tableName, CancellationToken cancellationToken = default) =>
+            await CreateTableInternalAsync(true, tableName, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Deletes a table in the storage account.
+        /// </summary>
+        /// <param name="tableName">The table name to create.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns></returns>
+        [ForwardsClientCalls]
+        public virtual Response DeleteTable(string tableName, CancellationToken cancellationToken = default) =>
+            DeleteTableInternalAsync(false, tableName, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Deletes a table in the storage account.
+        /// </summary>
+        /// <param name="tableName">The table name to create.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns></returns>
+        [ForwardsClientCalls]
+        public virtual async Task<Response> DeleteTableAsync(string tableName, CancellationToken cancellationToken = default) =>
+            await DeleteTableInternalAsync(true, tableName, cancellationToken).ConfigureAwait(false);
+
+        internal async Task<TableResponse> CreateTableInternalAsync(bool async, string tableName, CancellationToken cancellationToken = default)
+        {
+            if (async)
+            {
+                //using var scope = _clientDiagnostics.CreateScope("TableServiceClient.CreateTable");
+                //scope.Start();
+                return await _tableOperations.RestClient.CreateAsync(new TableProperties(tableName), null, new QueryOptions { Format = _format }, cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                //using var scope = _clientDiagnostics.CreateScope("TableServiceClient.CreateTable");
+                //scope.Start();
+                return _tableOperations.RestClient.Create(new TableProperties(tableName), null, new QueryOptions { Format = _format }, cancellationToken: cancellationToken);
+            }
+        }
+
+        internal async Task<Response> DeleteTableInternalAsync(bool async, string tableName, CancellationToken cancellationToken = default)
```

Comment:
Why helper method?

Reply:
Good question 😄 It isn't adding any value here.